File: launch_school/RB101-Foundations/small_problems/letter_swap.rb
=begin
-basically, the given string can be split into words
-each word can be mapped by swapping the first and last letter
-how does one do the swapping?
-assign the first and last letter to different variables
-then reassign those letters back to the string 
=end
def swap_one(word)
  word[0], word[-1] = word[-1], word[0]
  word
end

def swap(str)
  if str.length < 1
    str
  else
    words_arr = str.split(" ")
    words_arr.map! {|word| swap_one(word)}
    words_arr.join(' ')
  end
end

puts swap('Oh what a wonderful day it is') #== 'hO thaw a londerfuw yad ti si'
puts swap('Abcde') == 'ebcdA'
puts swap('a') == 'a'

File: launch_school/JS210-Fundamentals-of-Javascript/small_problems/easy_three/letter_swap.js
'use strict';

function swap(string) {
  let words = string.split(' ');

  let swappedWords = words.map(element => {
    let chars = element.split('');
    let firstChar = chars[0];
    let lastChar = chars[chars.length - 1];

    [chars[0], chars[chars.length - 1]] = [lastChar, firstChar];

    return chars.join('');
  });

  return swappedWords.join(' ');
}

swap('Oh what a wonderful day it is');  // "hO thaw a londerfuw yad ti si"
swap('Abcde');                          // "ebcdA"
swap('a');                              // "a"
